### PR TITLE
Convert ProjectsListAdapter to Kotlin

### PR DESCRIPTION
**Pull request by pAkalpa**
Fixes #4414 

**Description of the Change**

Convert ProjectsListAdapter.java to ProjectsListAdapter.kt

**Alternate Designs**

nil

**Release Notes**

### CODE REVIEW

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/ProjectsListAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,313 @@
+/*
+ * This file is part of BOINC.
+ * http://boinc.berkeley.edu
+ * Copyright (C) 2020 University of California
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/ProjectsListAdapter.kt:4]: Please change to 2022

File: android/BOINC/app/src/test/java/edu/berkeley/boinc/adapter/ProjectsListAdapterTest.kt [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        )
+        Assert.assertEquals(0, projectsListAdapter.getItemId(0))
+        Assert.assertEquals("Project 2", projectsListAdapter.getItem(1).project?.projectName
+        )
+        Assert.assertEquals(1, projectsListAdapter.getItemId(1))
+        Assert.assertEquals("Project 3", projectsListAdapter.getItem(2).project?.projectName
+        )
+        Assert.assertEquals(2, projectsListAdapter.getItemId(2))
+    }
+}
```
Comment (AenBleidd) [inline android/BOINC/app/src/test/java/edu/berkeley/boinc/adapter/ProjectsListAdapterTest.kt:89]: Please add new line

File: android/BOINC/app/src/test/java/edu/berkeley/boinc/adapter/ProjectsListAdapterTest.kt [hunk 3]
```diff
@@ -0,0 +1,89 @@
+package edu.berkeley.boinc.adapter
```
Comment (AenBleidd) [inline android/BOINC/app/src/test/java/edu/berkeley/boinc/adapter/ProjectsListAdapterTest.kt:1]: Please add copyright header

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/ProjectsListAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+                    ContextCompat.getColor(
+                        activity.applicationContext,
+                        color.transparent
+                    )
+                )
+            }
+        }
+        return vi
+    }
+}
```
Comment (AenBleidd) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/ProjectsListAdapter.kt:313]: Please add new line

**Review by AenBleidd — CHANGES_REQUESTED** [review]

**Comment by AenBleidd** [comment]
@pAkalpa, application if crashing when trying to open 'Projects' tab:
```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: edu.berkeley.boinc, PID: 31976
    java.lang.NullPointerException: null cannot be cast to non-null type kotlin.String
        at edu.berkeley.boinc.adapter.ProjectsListAdapter.getView(ProjectsListAdapter.kt:171)
        at android.widget.AbsListView.obtainView(AbsListView.java:2422)
        at android.widget.ListView.makeAndAddView(ListView.java:2067)
        at android.widget.ListView.fillDown(ListView.java:793)
        at android.widget.ListView.fillFromTop(ListView.java:855)
        at android.widget.ListView.layoutChildren(ListView.java:1810)
        at android.widget.AbsListView.onLayout(AbsListView.java:2219)
        at android.view.View.layout(View.java:23563)
        at android.view.ViewGroup.layout(ViewGroup.java:6452)
        at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
        at android.widget.LinearLayout.layoutHorizontal(LinearLayout.java:1818)
        at android.widget.LinearLayout.onLayout(LinearLayout.java:1584)
        at android.view.View.layout(View.java:23563)
        at android.view.ViewGroup.layout(ViewGroup.java:6452)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
        at android.view.View.layout(View.java:23563)
        at android.view.ViewGroup.layout(ViewGroup.java:6452)
        at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1103)
        at android.view.View.layout(View.java:23563)
        at android.view.ViewGroup.layout(ViewGroup.java:6452)
        at androidx.drawerlayout.widget.DrawerLayout.onLayout(DrawerLayout.java:1263)
        at android.view.View.layout(View.java:23563)
        at android.view.ViewGroup.layout(ViewGroup.java:6452)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
        at android.view.View.layout(View.java:23563)
        at android.view.ViewGroup.layout(ViewGroup.java:6452)
        at androidx.appcompat.widget.ActionBarOverlayLayout.onLayout(ActionBarOverlayLayout.java:536)
        at android.view.View.layout(View.java:23563)
        at android.view.ViewGroup.layout(ViewGroup.java:6452)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
        at android.view.View.layout(View.java:23563)
        at android.view.ViewGroup.layout(ViewGroup.java:6452)
        at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
        at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
        at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
        at android.view.View.layout(View.java:23563)
        at android.view.ViewGroup.layout(ViewGroup.java:6452)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
        at com.android.internal.policy.DecorView.onLayout(DecorView.java:868)
        at android.view.View.layout(View.java:23563)
        at android.view.ViewGroup.layout(ViewGroup.java:6452)
        at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:3863)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3313)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2277)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:9037)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1142)
        at android.view.Choreographer.doCallbacks(Choreographer.java:946)
        at android.view.Choreographer.doFrame(Choreographer.java:875)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1127)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:210)
        at android.os.Looper.loop(Looper.java:299)
        at android.app.ActivityThread.main(ActivityThread.java:8283)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:556)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1045)
```

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/ProjectsListAdapter.kt (hunk 1 above)
Comment (Vulpine05) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/ProjectsListAdapter.kt:4]: https, too?

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/ProjectsListAdapter.kt (hunk 4 above)
Comment (AenBleidd) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/ProjectsListAdapter.kt:313]: This new line is still missing

**Comment by AenBleidd** [comment]
There are 3 deprecations. It would be nice to fix them as well.
image

**Comment by pAkalpa** [comment]
@AenBleidd 2 Checks Failed in CI but App Works and build fine in local build. 
image
To replace the Deprecated code I referred to this official Documentation ([https://developer.android.com/jetpack/androidx/releases/activity#1.4.0-alpha01](url))

**Comment by AenBleidd** [comment]
Please find the reason why it's failing and if possible - fix it.
Btw, to have successful build on CI is more important than have all deprecations fixed, that is why I moved fix of deprecations to a separate ticket.

**Comment by pAkalpa** [comment]
@AenBleidd I added androidx.activity dependency. Let's see if is it gonna fail again. If CI fails again  I will revert changes back to the previous deprecated version.

**Review by AenBleidd — APPROVED** [review]

**Pull request merged**